### PR TITLE
Uwp download

### PR DESCRIPTION
**Pull request by codergrl**
@nCastle1 please review
- added download and sync functionality to UWP
- updated offline mode to show proper error messages when no map is available

### CODE REVIEW

File: src/DataCollection.Shared/Properties/Settings.cs [hunk 1]
```diff
@@ -77,17 +77,17 @@ public static Settings Default
                         // create stream and deserialize into a Settings object
                         var stream = typeof(Settings).Assembly.GetManifestResourceStream(streamPath);
                         _instance = DeserializeSettings(stream);
+                        stream.Dispose();
```
Comment (nCastle1) [inline src/DataCollection.Shared/Properties/Settings.cs]: Could this be a using?

File: src/DataCollection.Shared/ViewModels/AuthViewModel.cs [hunk 2]
```diff
@@ -62,22 +61,14 @@ public AuthViewModel(string webmapURL, string arcGISOnlineURL, string appClientI
...
-                    {
-                        // login user if the status code for the web map is OK
-                        if (response.StatusCode == HttpStatusCode.OK)
-                        {
-                            LoginCommand.Execute(null);
-                        }
-                    }
-                }
-                catch { /* Leave the user logged off if device is offline */ }
+                // if device is online, login user automatically
```
Comment (nCastle1) [inline src/DataCollection.Shared/ViewModels/AuthViewModel.cs:64]: Looks like the indentation is a it off here

File: src/DataCollection.Shared/ViewModels/MainViewModel.cs [hunk 3]
```diff
@@ -27,22 +27,50 @@
...
+        // will throw if another platform is added without handling this 
+        throw new NotImplementedException();
+#endif
 
         public MainViewModel()
         {
             _webMapURL = Settings.Default.WebmapURL;
-            DownloadPath = Directory.Exists(Settings.Default.DownloadPath) ? Settings.Default.DownloadPath : null;
+
+
```
Comment (nCastle1) [inline src/DataCollection.Shared/ViewModels/MainViewModel.cs]: Looks like there's an extra blank line

File: src/DataCollection.UWP/MainPage.xaml [hunk 4]
```diff
@@ -73,22 +75,124 @@
...
                 <AppBarButton
+                    x:Uid="DeleteMapButton"
+                    Command="{x:Bind MainViewModel.DeleteOfflineMapCommand, Mode=OneWay}"
+                    Visibility="{x:Bind MainViewModel.OfflineMap, Mode=OneWay, Converter={StaticResource NullToVisibilityConverter}}">
+                    <AppBarButton.Icon>
+                        <FontIcon FontFamily="Segoe MDL2 Assets" Glyph="&#xE74D;" />
+                    </AppBarButton.Icon>
+                </AppBarButton>
+
+
```
Comment (nCastle1) [inline src/DataCollection.UWP/MainPage.xaml]: Looks like there's some extra whitespace. Might be worth adding a comment to be consistent with the rest of the buttons

File: src/DataCollection.UWP/MainPage.xaml [hunk 5]
```diff
@@ -127,6 +231,55 @@
...
+                    x:Uid="SynchingMessage"
+                    Grid.Row="1"
+                    Margin="0,0,0,20"
+                    HorizontalAlignment="Center" />
+                <Button
+                    Grid.Row="2"
+                    Width="70"
+                    HorizontalAlignment="Center"
+                    Command="{x:Bind MainViewModel.SyncViewModel.CancelSyncCommand}"
+                    Content="Cancel" />
```
Comment (nCastle1) [inline src/DataCollection.UWP/MainPage.xaml]: Looks like this needs to be localized

File: src/DataCollection.UWP/Views/DownloadView.xaml [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+                <Button
+                    x:Uid="DownloadButton"
+                    Margin="20,0,0,0"
+                    Command="{Binding DownloadViewModel.DownloadMapCommand}"
+                    CommandParameter="{Binding ElementName=DownloadUserControl, Path=VisibleArea}" />
+            </StackPanel>
+        </Grid>
+
+        <!--  Progress bar area for download  -->
+        <StackPanel
```
Comment (nCastle1) [inline src/DataCollection.UWP/Views/DownloadView.xaml:109]: I recommend setting a background color on the progress view. It currently looks fine without on light mode, but on dark mode the text changes color and can't be seen over the map. Might be an opportunity to incorporate some more acrylic :) 

Screen Shot 2019-05-06 at 6 21 50 PM

File: src/DataCollection.UWP/MainPage.xaml [hunk 7]
```diff
@@ -73,22 +75,124 @@
...
+                    Visibility="{x:Bind MainViewModel.ConnectivityMode, Mode=OneWay, Converter={StaticResource ConnectivityModeToVisibilityConverter}, ConverterParameter=Inverse}">
+                    <AppBarButton.Icon>
+                        <FontIcon FontFamily="Segoe MDL2 Assets" Glyph="&#xEC3F;" />
+                    </AppBarButton.Icon>
+                </AppBarButton>
+
+                <AppBarSeparator Visibility="{x:Bind MainViewModel.OfflineMap, Mode=OneWay, Converter={StaticResource NullToVisibilityConverter}}" />
+
+                <!--  Working offline button  -->
+                <AppBarButton x:Uid="WorkingOfflineButton" Visibility="{x:Bind MainViewModel.ConnectivityMode, Mode=OneWay, Converter={StaticResource ConnectivityModeToVisibilityConverter}, ConverterParameter=Inverse}">
```
Comment (nCastle1) [inline src/DataCollection.UWP/MainPage.xaml:120]: I would recommend setting `IsEnabled` false; it is a little confusing that it is a button that doesn't do anything.

**Review by nCastle1 — APPROVED** [review]
Looks great! Only a few comments

File: src/DataCollection.UWP/Views/DownloadView.xaml (hunk 6 above)
Comment (codergrl) [reply src/DataCollection.UWP/Views/DownloadView.xaml:109]: This and many more styling issues are being taken care of in the next PR.

**Pull request merged**